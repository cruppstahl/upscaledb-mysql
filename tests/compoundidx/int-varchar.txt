DROP TABLE IF EXISTS test;
CREATE TABLE test (id INT NOT NULL PRIMARY KEY,
    c1 INTEGER NOT NULL,
    c2 TEXT NOT NULL,
    INDEX name (c1, c2(30))
) ENGINE=upscaledb;
INSERT INTO test VALUES(1, 10, "LAST1");
INSERT INTO test VALUES(2, 20, "LAST2");
INSERT INTO test VALUES(3, 30, "LAST3");
INSERT INTO test VALUES(4, 40, "LAST4");
INSERT INTO test VALUES(5, 50, "LAST5");
SELECT * FROM test;
SELECT * FROM test WHERE id >= 3;
SELECT * FROM test WHERE c2 = "LAST3";
SELECT * FROM test WHERE c1 = 30;
UPDATE test SET c2 = "LAST1" WHERE id = 5;
SELECT * FROM test;
UPDATE test SET c2 = "LAST55" WHERE id = 5;
SELECT * FROM test;
UPDATE test SET c1 = 51 WHERE id = 5;
SELECT * FROM test;
DELETE FROM test WHERE id = 1;
SELECT * FROM test;
