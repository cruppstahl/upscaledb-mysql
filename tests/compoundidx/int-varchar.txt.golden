--------------
DROP TABLE IF EXISTS test
--------------

--------------
CREATE TABLE test (id INT NOT NULL PRIMARY KEY,
    c1 INTEGER NOT NULL,
    c2 TEXT NOT NULL,
    INDEX name (c1, c2(30))
) ENGINE=upscaledb
--------------

--------------
INSERT INTO test VALUES(1, 10, "LAST1")
--------------

--------------
INSERT INTO test VALUES(2, 20, "LAST2")
--------------

--------------
INSERT INTO test VALUES(3, 30, "LAST3")
--------------

--------------
INSERT INTO test VALUES(4, 40, "LAST4")
--------------

--------------
INSERT INTO test VALUES(5, 50, "LAST5")
--------------

--------------
SELECT * FROM test
--------------

id	c1	c2
1	10	LAST1
2	20	LAST2
3	30	LAST3
4	40	LAST4
5	50	LAST5
--------------
SELECT * FROM test WHERE id >= 3
--------------

id	c1	c2
3	30	LAST3
4	40	LAST4
5	50	LAST5
--------------
SELECT * FROM test WHERE c2 = "LAST3"
--------------

id	c1	c2
3	30	LAST3
--------------
SELECT * FROM test WHERE c1 = 30
--------------

id	c1	c2
3	30	LAST3
--------------
UPDATE test SET c2 = "LAST1" WHERE id = 5
--------------

--------------
SELECT * FROM test
--------------

id	c1	c2
1	10	LAST1
2	20	LAST2
3	30	LAST3
4	40	LAST4
5	50	LAST1
--------------
UPDATE test SET c2 = "LAST55" WHERE id = 5
--------------

--------------
SELECT * FROM test
--------------

id	c1	c2
1	10	LAST1
2	20	LAST2
3	30	LAST3
4	40	LAST4
5	50	LAST55
--------------
UPDATE test SET c1 = 51 WHERE id = 5
--------------

--------------
SELECT * FROM test
--------------

id	c1	c2
1	10	LAST1
2	20	LAST2
3	30	LAST3
4	40	LAST4
5	51	LAST55
--------------
DELETE FROM test WHERE id = 1
--------------

--------------
SELECT * FROM test
--------------

id	c1	c2
2	20	LAST2
3	30	LAST3
4	40	LAST4
5	51	LAST55
